### PR TITLE
Add better clarify in UG

### LINKED ISSUE #182: [PE-D][Tester A] Delete Command returns the wrong error message when user enters a very large positive number
If I entered a very positive number, the error returned is different than what is expected and should show `The event index provided is invalid` since the command is a valid command.

### Expected Consistent Outcome

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: Acerizm/ped#1

### LINKED ISSUE #222: [PE-D][Tester A] edit command with given `t/tag` command deletes all existing tags and replaced the first tag with the new tag
The edit command should not delete all tags and should instead allow the user to edit the specific tag. The user may become confused as there can be many tags assigned to the event but the edit command can completely wipe all the tags assigned before. 

Maybe the edit command can select which tag to edit! 

### Result after running `edit 2 t/testTag4`

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: Acerizm/ped#5

### PR DESCRIPTION
**Pull request by engenhui1999**
- Add better clarity to explain that editing of tags will remove all existing tags.
- Add better clarity to safeguard against extreme user inputs beyond unsigned integers.
- Closes #222 
- Closes #182

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -174,7 +174,7 @@ Adds a new event.
 
 **Note**:
 - `NAME` should only contain alphanumeric characters and spaces.
-- `RATE` must be a positive number.
+- `RATE` must be a positive number and below 1,000,000.
```
Comment (glozxi) [inline docs/UserGuide.md:177]: Should `below 1,000,000` be in another sentence to be more consistent with the other commands in the UG?

**Review by glozxi — CHANGES_REQUESTED** [review]
LGTM with a nit

**Review by ExtraShotLatte — COMMENTED** [review]
LGTM, just some nits about the where we place things! 👍

**Review by glozxi — APPROVED** [review]
LGTM 😃

**Pull request merged**